### PR TITLE
Challenge

### PR DESCRIPTION
**Pull request by divyesh2506**
- [x]  Add a way to delete companies - Adds a soft delete option

- [x] Add Custom Validation to company email

- [x] Add City State To Companies from Zip Code

- [x] Include a way to set a brand color

- [x] Include test coverage for all your changes

### CODE REVIEW

**Comment by divyesh2506** [comment]
For additional challenge a docker setup would be a good challenge

File: app/models/company.rb [hunk 1]
```diff
@@ -1,5 +1,20 @@
 class Company < ApplicationRecord
   has_rich_text :description
+  default_scope { where(active: true) }
```
Comment (Sushanthbc) [inline app/models/company.rb:3]: Default scope — Interesting. I always like my model to give the entire dataset. If I would like to filter the data, I go with the named scope. I feel using `unscoped` for fetching active and inactive companies can be a strain.

Also, performing deletion operations on the associated models can be a little tricky.

File: app/models/company.rb [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+  VALID_EMAIL_REGEX =  /[\w\-\.]+(@getmainstreet\.com)/
+  validates_format_of :email, with: VALID_EMAIL_REGEX, allow_blank: true, if: :email_changed?
+  before_save :update_locality, if: :zip_code_changed?
```
Comment (Sushanthbc) [inline app/models/company.rb:6]: Nice use of callback

File: app/models/company.rb [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
 
+  def delete
+    # Using update attribute helps avoid validation check on existing 
+    update_attribute('active', false)
+  end
 
+  def update_locality
```
Comment (Sushanthbc) [inline app/models/company.rb:13]: 💯

File: app/controllers/companies_controller.rb [hunk 4]
```diff
@@ -32,6 +32,14 @@ def update
     end
   end  
 
+  def destroy
+    if @company.delete
+      redirect_to companies_path, notice: "Deleted"
+    else
+      redirect_to companies_path, notice: @company.errors.full_messages.join(", ")
```
Comment (Sushanthbc) [inline app/controllers/companies_controller.rb:39]: Nice feedback to end-user, while deleting 👍

File: app/views/companies/_form.html.erb [hunk 5]
```diff
@@ -1,80 +1,96 @@
...
+    </div>
+  </div>
+
+  <div class="row my-lg-3">
+    <div class="col-lg-3">
+      <label><strong>Brand Color</strong><br><small class="text-muted">Brand color of company</small></label>
+    </div>
+    <div class="col-lg offset-lg-1">
+      <div data-controller="services" class="form-group">
+        <%= f.color_field :color, class: "form-control" %>
```
Comment (Sushanthbc) [inline app/views/companies/_form.html.erb:83]: I like how you have used color as the input field. However, We are looking more in line with the usage of Javascript, especially Stimulusjs

https://github.com/main-street/code-challenge#-include-a-way-to-set-a-brand-color

Yes, I like the idea of persisting the value between each server request.

**Review by Sushanthbc — COMMENTED** [review]
Appreciate taking the time to do the assignment — we will follow up over email.

**Pull request closed without merging**